entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/ValueHelper.java

```diff
@@ -81,6 +83,10 @@ public boolean isMap(final ValueType valueType) {
     return MAP_TYPES.contains(valueType);
   }
 
+  public boolean isNull(final Value value) {
+    return NULL_VALUE.equalsIgnoreCase(value.getString());
```

Comment:
What's the use case of the null case in QS? Who sets the null case value type?

Reply:
Callers looking for things to not be null, for example :) Previously the only way to do that was the magic string "null", and that means we can't have a value actually named null. Also problematic for fields that aren't typed string. Put slightly differently, the purpose was to avoid putting implementation defaults/details on the caller.

Reply:
I think I have hit this. Basically, the caller (gateway service) asks for say column1, column2, and there is a filter on `column2 != null`, the callee(entity service) would to still populate in some values for column2's `null` row

Currently, we put an empty string https://github.com/hypertrace/entity-service/blob/9e4cfdc4ff0b1e6bb5e68cf90f567e1a977ee193/entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/response/DocumentConverterImpl.java#L43 in entity-service

How should the `NEQ` null filtering happen in that case? The caller (gateway service) would have to take care of the `NEQ` null filtering in that case

Reply:
I'm not quite sure I'm following - if there is a filter provided from GW on NEQ null, why are any results being returned that require populating a default value for null? There should never be a case where the caller has to do any filtering (so if there is, let's identify them)